go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -205,6 +205,20 @@ func (st *symtab) AllTables() []*table {
 	return tables
 }
 
+// AllVschemaTableNames returns an ordered list of all current vschema tables.
+func (st *symtab) AllVschemaTableNames() []*vindexes.Table {
+	if len(st.tableNames) == 0 {
+		return nil
+	}
+	tables := make([]*vindexes.Table, 0, len(st.tableNames))
+	for _, tname := range st.tableNames {
+		if vschemaTable := st.tables[tname].vschemaTable; vschemaTable != nil {
```

Comment:
vschemaTable  is nil for `information_schema` tables, so we ignore them here. these tables seem to appear correctly anyway.
Anyhow, I found this https://github.com/vitessio/vitess/blob/acb8c2f762faea1637e62790bcd934d96b47e538/go/vt/vtgate/planbuilder/testdata/from_cases.txt#L2243-L2278 - table names are bad. Maybe there is a reason for this, I don't know...

Reply:
This is due to https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/engine/join.go#L241
Could be fixed in another PR since it involves lots of cases.